lib/session.js

```diff
@@ -166,10 +156,13 @@ module.exports = class Session {
     return this[originalHash] !== this[hash]()
   }
 
-  static restore (request, idGenerator, cookieOpts, secret, prevSession) {
-    const restoredSession = new Session(request, idGenerator, cookieOpts, secret, prevSession)
+  static restore (request, idGenerator, cookieOpts, secret, prevSession, sessionId) {
+    const restoredSession = new Session(request, idGenerator, cookieOpts, secret, prevSession, sessionId)
     const restoredCookie = new Cookie(cookieOpts)
-    restoredCookie.expires = new Date(prevSession.cookie.expires)
+    if (prevSession.cookie.expires) {
+      // Need to parse as Date
+      restoredCookie.expires = new Date(prevSession.cookie.expires)
+    }
```

Comment:
This is a bugfix! Otherwise `Session.restore()` will set `expires`, even when it's not set.